go/arrow/csv/reader.go

```diff
@@ -767,7 +767,7 @@ func (r *Reader) parseList(field array.Builder, str string) {
 
 func (r *Reader) parseBinaryType(field array.Builder, str string) {
 	// specialize the implementation when we know we cannot have nulls
-	if str != "" && r.isNull(str) {
+	if r.isNull(str) {
 		field.AppendNull()
 		return
 	}
```

Comment:
Do we currently have a test that verifies you can still get an empty string if it is not in the list of Null values? / `stringsCanBeNull` == `false` ?

Reply:
I think not, but happy to add one here

Reply:
Added two new test cases for this